plasmapy/atomic/tests/test_atomic.py

```diff
@@ -660,60 +714,65 @@ def test_isotopic_abundance():
     with pytest.raises(ValueError):
         isotopic_abundance('Og-2')
 
-    # Check that the sum of isotopic abundances for each element is one
-
-    for atomic_numb in range(1, 119):
 
-        element = atomic_number(atomic_numb)
-        isotopes = common_isotopes(element)
+isotopic_abundance_elements = (
+    atomic_number(atomic_numb) for atomic_numb in range(1, 119))
+isotopic_abundance_isotopes = (
+    common_isotopes(element) for element in isotopic_abundance_elements)
+isotopic_abundance_sum_table = [
+    (element, isotopes)
+    for element, isotopes in
+    zip(isotopic_abundance_elements, isotopic_abundance_isotopes)
+    if isotopes]
```

Comment:
I'm curious whether this wouldn't run faster with a generator instead of a list, given that at no point here do we use more than a single element from this list, so there's no need to keep the whole list in memory. I wasn't able to show an explicit variation in speed over noise this way, though. The code is quite fine, I'm just thinking aloud (well, the keyboard is loud).

Reply:
Good suggestion!

Reply:
Haha, I wasn't really suggesting changing this one, but yeah, we might as well go with generators here :laughing: